From 061ffb64cfb42ca866a2250ec63613df3a18f916 Mon Sep 17 00:00:00 2001
From: Guillaume Betous <guillaumex.betous@intel.com>
Date: Fri, 3 Oct 2014 16:25:08 +0200
Subject: Allow custom make_recovery_patch

When using BOARD_CUSTOM_MKBOOTIMG, one may also need
a different make_recovery_patch in order to generate a patch
with specific pre/post processing.
Adding the variable BOARD_CUSTOM_MAKE_RECOVERY_PATCH
in the same spirit.

Change-Id: I5279f7a2ab5e88ee37f0b6def249832de5a51345
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-1988
Signed-off-by: Guillaume Betous <guillaumex.betous@intel.com>
Category: device enablement
Domain: PUPDR.Recovery
Origin: internal
Upstream-Candidate: yes
---
 core/Makefile  |    2 +-
 core/config.mk |    5 +++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/core/Makefile b/core/Makefile
index b3a4c4b..c97995b 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -1419,7 +1419,7 @@ ifneq ($(OEM_THUMBPRINT_PROPERTIES),)
 	$(hide) echo "oem_fingerprint_properties=$(OEM_THUMBPRINT_PROPERTIES)" >> $(zip_root)/META/misc_info.txt
 endif
 	$(call generate-userimage-prop-dictionary, $(zip_root)/META/misc_info.txt)
-	$(hide) ./build/tools/releasetools/make_recovery_patch $(zip_root) $(zip_root)
+	$(hide) $(MAKE_RECOVERY_PATCH) $(zip_root) $(zip_root)
 	@# Zip everything up, preserving symlinks
 	$(hide) (cd $(zip_root) && zip -qry ../$(notdir $@) .)
 	@# Run fs_config on all the system, vendor, boot ramdisk,
diff --git a/core/config.mk b/core/config.mk
index bfa5610..aa27e43 100644
--- a/core/config.mk
+++ b/core/config.mk
@@ -389,6 +389,11 @@ MKBOOTIMG := $(HOST_OUT_EXECUTABLES)/mkbootimg$(HOST_EXECUTABLE_SUFFIX)
 else
 MKBOOTIMG := $(BOARD_CUSTOM_MKBOOTIMG)
 endif
+ifeq (,$(strip $(BOARD_CUSTOM_MAKE_RECOVERY_PATCH)))
+MAKE_RECOVERY_PATCH := ./build/tools/releasetools/make_recovery_patch
+else
+MAKE_RECOVERY_PATCH := $(BOARD_CUSTOM_MAKE_RECOVERY_PATCH)
+endif
 MKYAFFS2 := $(HOST_OUT_EXECUTABLES)/mkyaffs2image$(HOST_EXECUTABLE_SUFFIX)
 APICHECK := $(HOST_OUT_EXECUTABLES)/apicheck$(HOST_EXECUTABLE_SUFFIX)
 FS_GET_STATS := $(HOST_OUT_EXECUTABLES)/fs_get_stats$(HOST_EXECUTABLE_SUFFIX)
-- 
1.7.10.4

